File: bootcamp/src/main/java/io/nextree/board/aggregate/Board.java
package io.nextree.board.aggregate;

import io.nextree.Util.IdName;
import io.nextree.board.aggregate.sdo.BoardCdo;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Board {
    //
    private String boardId;
    private String title;
    private IdName user;
    private long regDate;
    private long modDate;
    private int views;
    private String content;

    public Board(BoardCdo command) {
        //
        this.title = command.getTitle();
        this.content = command.getContent();
        this.user = command.getUser();

        this.boardId = UUID.randomUUID().toString();
        this.regDate = System.currentTimeMillis();
        this.modDate = 0;
        this.views = 0;
    }

    public void updateData(Board updateBoard) {
        this.title = updateBoard.getTitle();
        this.content = updateBoard.getContent();
        this.modDate = System.currentTimeMillis();
    }
}

File: bootcamp-app/src/aggregate/Board.ts
import {IdName, newIdName} from "./vo/IdName";

export interface Board {
    boardId: string;
    title: string;

    user: IdName;
    regDate: number;
    modDate: number;
    views: number;
    content: string;
}
export const newBoard = (boardId='', title='', user=newIdName(), regDate=0, modDate=0, views=0, content=''):Board => {
    return {boardId, title, user, regDate, modDate, views, content}
};
